torch/utils/_pytree.py

```diff
@@ -181,6 +178,28 @@ def _register_pytree_node(
     SUPPORTED_SERIALIZED_TYPES[cls] = serialize_node_def
     SERIALIZED_TYPE_TO_PYTHON_TYPE[serialized_type_name] = cls
 
+    import torch
+
+    if torch._running_with_deploy():
+        warnings.warn("C++ pytree utilities do not work with torch::deploy.")
+        return
```

Comment:
- Importing cxx_pytree should fail when running_with_deploy()
- Then, we don't need to warn here: users will see the warning if they are using deploy, and importing cxx_pytree.